src/cloudant/index.py

```diff
@@ -262,3 +267,201 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes. A SearchIndex object is
+    instantiated with a reference to a DesignDocument and is typically used as
+    part of the :class:`~cloudant.design_document.DesignDocument`
+    search index management API.
+
+    :param DesignDocument ddoc: DesignDocument instance used in part to
+        identify the search index.
+    :param str search_index_name: Name used in part to identify the index.
+    :param str search_index_func: Javascript search index function.
```

Comment:
I like the new names (index_name and search_func) but the docstring needs to be fixed to match them.